operators/pkg/instctrl/cloudinit.go

```diff
@@ -45,47 +46,82 @@ const (
 func (r *InstanceReconciler) EnforceCloudInitSecret(ctx context.Context) error {
 	log := ctrl.LoggerFrom(ctx)
 
-	// Retrieve the WebDav credentials.
-	user, password, err := r.GetWebDavCredentials(ctx)
-	if err != nil {
-		log.Error(err, "unable to get webdav credentials")
-		return err
-	}
-	log.V(utils.LogDebugLevel).Info("webdav credentials correctly retrieved")
+	var retErr error
+	var server string
+	var path string
 
 	// Retrieve the public keys
-	publicKeys, err := r.GetPublicKeys(ctx)
-	if err != nil {
-		log.Error(err, "unable to get public keys")
-		return err
+	publicKeys, retErr := r.GetPublicKeys(ctx)
+	if retErr != nil {
+		log.Error(retErr, "unable to get public keys")
+		return retErr
```

Comment:
would revert this

Reply:
I don't think I fully understand what you are saying. I've changed the name of the variable back to ```err``` and placed this piece of code after the get of the nfs secret like it was after the ```r.GetWebDavCredentials(ctx)```.